test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -2136,18 +2136,18 @@ protected static TransportVersion minimumTransportVersion() throws IOException {
     }
 
     protected static TransportVersion getTransportVersionWithFallback(
-        Object versionField,
+        String versionField,
         Object transportVersionField,
         Supplier<TransportVersion> fallbackSupplier
     ) {
-        if (transportVersionField instanceof Integer transportVersionId) {
-            return TransportVersion.fromId(transportVersionId);
+        if (transportVersionField instanceof Number transportVersionId) {
+            return TransportVersion.fromId((int) transportVersionId);
```

Comment:
This is still assuming its going to be an `Integer`. You should use `Number.intValue` instead